### ISSUE TITLE
📎 Vue noFloatingPromises not "checked" in vue script tag with lang="ts"

### ISSUE THREAD

**Issue by bipedal-eel** [biomejs/biome#10304]
### Description

The vue script tag with lange="ts" does not respect noFloatingPromises rule or what I'd assume is that the types domain is not taking vue files into account

```vue
<script lang="ts">
  const a = async () => {
    // this should show a diagnostic
  	Promise.resolve();
  };
  
  const b = () => {
    // this should show a diagnostic
  	a();
  };
</script>
```

[vue](https://biomejs.dev/playground/?lintRules=noFloatingPromises&tab=syntax&pane=Diagnostics&code=PABzAGMAcgBpAHAAdAAgAGwAYQBuAGcAPQAiAHQAcwAiAD4ACgAgACAAYwBvAG4AcwB0ACAAYQAgAD0AIABhAHMAeQBuAGMAIAAoACkAIAA9AD4AIAB7AAoAIAAgAAkAUAByAG8AbQBpAHMAZQAuAHIAZQBzAG8AbAB2AGUAKAApADsACgAgACAAfQA7AAoAIAAgAAoAIAAgAGMAbwBuAHMAdAAgAGIAIAA9ACAAKAApACAAPQA%2BACAAewAKACAAIAAJAGEAKAApADsACgAgACAAfQA7AAoAPAAvAHMAYwByAGkAcAB0AD4A&language=vue)

[typescript](https://biomejs.dev/playground/?lintRules=noFloatingPromises&tab=syntax&pane=Diagnostics&code=bABlAHQAIABhACAAPQAgAGEAcwB5AG4AYwAgACgAKQAgAD0APgAgAHsACgAgACAAUAByAG8AbQBpAHMAZQAuAHIAZQBzAG8AbAB2AGUAKAApADsACgB9ADsACgAKAGMAbwBuAHMAdAAgAGIAIAA9ACAAKAApACAAPQA%2BACAAewAKACAAIABhACgAKQA7AAoAfQA7AA%3D%3D&language=ts)

**Comment by ematipico** [comment]
Please open a new issue using the bug report template, sharing the proper information
